### ISSUE TITLE
About the network

### ISSUE THREAD

**Issue by YongshengDong** [nibtehaz/MultiResUNet#15]
https://github.com/nibtehaz/MultiResUNet/blob/0190db6c1f3725bcda42f50e64006ff9d91bc38b/MultiResUNet.py#L136

Hi, 
I have some confusion about the network:
1. Generally, BN layer is followed by Activation, this can be found in official Resnet's implement, I want to know your ideas, thx.
2. Why you place anothor Activation and BN layer after every feature fusion(add/concat), is it based on experimental results?

**Comment by nibtehaz** [comment]
Hello @YongshengDong . Thank you for your interest in our project.

Regarding your queries,

1. Yes the ordering is a bit debatable as both are used. Particularly, I prefer the following idea, here I am quoting the answer (https://stackoverflow.com/a/40295999) from this question in Stack Overflow (https://stackoverflow.com/questions/39691902/ordering-of-batch-normalization-and-dropout) kudos to Zhongyu Kuang

> In the Ioffe and Szegedy 2015, the authors state that "we would like to ensure that for any parameter values, the network always produces activations with the desired distribution". So the Batch Normalization Layer is actually inserted right after a Conv Layer/Fully Connected Layer, but before feeding into ReLu (or any other kinds of) activation. See this video at around time 53 min for more details.
>
>As far as dropout goes, I believe dropout is applied after activation layer. In the dropout paper figure 3b, the dropout factor/probability matrix r(l) for hidden layer l is applied to it on y(l), where y(l) is the result after applying activation function f.
>
>So in summary, the order of using batch normalization and dropout is:
>
>-> CONV/FC -> BatchNorm -> ReLu(or other activation) -> Dropout -> CONV/FC ->


2. Yes it was based on initial experimental results as they seemed to improve the results a bit.
